rpc/client_server_test.go

```diff
@@ -196,3 +204,28 @@ func TestOrdersSubscription(t *testing.T) {
 	// The WaitGroup signals that AddOrder was called on the server-side.
 	wg.Wait()
 }
+
+func TestHeartbeatSubscription(t *testing.T) {
+	ctx := context.Background()
+
+	// Set up the dummy handler with a subscribeToHeartbeatHandler
+	wg := &sync.WaitGroup{}
```

Comment:
Can we use a channel to signal when the `subscribeToHeartbeatHandler` was called instead of a `sync.WaitGroup`? It would improve the failure mode a lot. Right now the test will hang for 10 minutes (the default test timeout in Go) and eventually timeout, but we could easily fail faster. Something like:

```go
rpcHandlerCalled := make(chan struct{})
rpcHandler := &dummyRPCHandler{
    subscribeToHeartbeatHandler: func(ctx context.Context) (*rpc.Subscription, error) {
        rpcHandlerCalled <- struct{}{}
        return nil, nil
    },
}

// The select statement allows us to timeout sooner than the default test timeout.
select {
    case <-time.After(5 * time.Second):
        t.Error("timed out waiting for rpcHandler to be called")
    case <-rpcHandlerCalled:
        break
}
```